crates/binjs_es6/src/scopes.rs

```diff
@@ -55,16 +53,23 @@ impl AnnotationVisitor {
     }
 
     fn push_free_names(&mut self) {
-        self.free_names_in_function_stack.push(HashSet::new());
+        self.free_names_in_block_stack.push(HashMap::new());
     }
-    fn pop_free_names(&mut self, bindings: &[&HashSet<String>]) {
-        let mut free_names_in_current_function = self.free_names_in_function_stack.pop().unwrap();
-        for name in free_names_in_current_function.drain() {
+    fn pop_free_names(&mut self, bindings: &[&HashSet<String>], function_scope: bool) {
```

Comment:
Nit: Any chance you could replace `function_scope` with an `enum`? This would make things a bit more readable for callers.

Also, if it's a  boolean, I'd rather call it `is_function_scope`.